images/cdi-http-import-server/entrypoint.sh

```diff
@@ -37,6 +38,12 @@ fedora-cloud) CONVERT_PATH=$FEDORA_IMAGE_PATH ;;
     ;;
 esac
 
+if [ $IPV6 == "true" ]; then
```

Comment:
I'm almost sure you don't need any of this; I think you can just blindly bind to `[::]` which will accept also IPv4 traffic.

@oshoval I know we've discussed this in the past, and I lost track of the final 'decision' ... In the production code we're binding only to IPv6, or checking which family to bind to ?

Reply:
about this specific line i wrote in the chat the same, but i dont know if it was tested / works in case it tested.

in the production code of kubevirt we use go net lib for the sockets that have a wrapper which will listen always to ipv6 even if you asked 0.0.0.0, unless you specifically asked it not to, and we changed it to be listen explicitly to the one that we need to. if thats what you mean.

Reply:
So we listen to ipv4 **or** ipv6, depending on the IP ?

Do you happen to remember why we changed it to work like that ? It would be a lot simpler to just bind to one (and this is not production code, so the requirements may be somewhat laxer). 

Thanks for pitching in.

Reply:
https://github.com/kubevirt/kubevirt/pull/3396
correction, we bind to [::] explicitly (unless we cant), this was already the situation before, we just made it explicit (and not hide in inner golang net lib)

but here it might have its own rules ? i dont know by heart the cdi importer code, if we indeed can listen to [::] its the best of course.

Reply:
Listening on `[::]` only will cause ISCSI PVC tests to fail if the target pod has IPv4 address
I dont like the idea of enableing the mapping between in this case.
It better to create another target pod or configure it to work with IPv6 on demand like what I presented in this PR using an env variable.